projects/igniteui-angular/src/lib/grids/grid-mrl-navigation.service.ts

```diff
@@ -426,9 +426,15 @@ export class IgxGridMRLNavigationService extends IgxGridNavigationService {
         return '.igx-grid__mrl-block';
     }
 
+    public getParentUnpinnedIndex(visibleColumnIndex: number) {
+        const column = this.grid.unpinnedColumns.find((col) => col.columnLayout && col.visibleIndex === visibleColumnIndex);
+        return this.grid.pinnedColumns.length ? this.grid.unpinnedColumns.filter((c) => c.columnLayout).indexOf(column) :
+            visibleColumnIndex;
+    }
+
     protected performHorizontalScrollToCell(rowIndex, visibleColumnIndex, isSummary = false) {
         const col = this.grid.columns.find(x => !x.columnGroup && x.visibleIndex === visibleColumnIndex);
-        const unpinnedIndex = this.getColumnUnpinnedIndex(col.parent.visibleIndex);
+        const unpinnedIndex = this.getParentUnpinnedIndex(col.parent.visibleIndex);
```

Comment:
Instead of exposing another function that is very similar to the base one, can't we get the index directly from the igxFor, like it is done in performTab/performShiftTabKey functions, for example:

  const hScroll = this.horizontalScroll(rowIndex);
  const scrIndex = hScroll.igxForOf.indexOf(col.parent);
  hScroll.scrollTo(scrIndex);